src/test/java/ca/qaguru/oranghrmbatch27/pages/PayGradePage.java

```diff
@@ -0,0 +1,86 @@
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+import java.util.List;
+
+public class PayGradePage extends PageBase {
+
+    public PayGradePage(WebDriver _driver) {
+        super(_driver);
+        PageFactory.initElements(_driver, this);
+    }
+
+    private String adminMenuPath = "//span[text()=\"Admin\"]";
+    private String jobTopMenuPath = "//span[normalize-space()=\"Job\"]";
+    private String payGradeDropDownMenuPath = "//a[text()=\"Pay Grades\"]";
+    private String addButtonPath = "//button[normalize-space()=\"Add\"]";
+    private String payGradeNameTextboxPath = "//div[@class=\"oxd-form-row\"]//input";
+    private String saveButtonPath = "//button[normalize-space()=\"Save\"]";
+    private String requiredFieldPath = "//div[@class=\"oxd-form-row\"]//span";
+    private String successMessagePath = "//div[@class=\"oxd-toast-content oxd-toast-content--success\"]";
+    private String cancelButtonXpath = "//button[normalize-space()=\"Cancel\"]";
+    private String noOfRecordsXpath = "//div[@class=\"orangehrm-horizontal-padding orangehrm-vertical-padding\"]//span";
+    private String yesDeleteButtonXpath = "//div[@class=\"orangehrm-modal-footer\"]//button[2]";
+    private String xpathSuccessMessage = "//div[@class=\"oxd-toast oxd-toast--success oxd-toast-container--toast\"]";
+
+    @FindBy(xpath = "//div[@class=\"oxd-table-body\"] //div[@class=\"oxd-table-cell oxd-padding-cell\"][2]")
+    List<WebElement> listOfRecordNames;
+
+    @FindBy(xpath = "//i[@class=\"oxd-icon bi-trash\"]")
+    List<WebElement> delete;
+
+    public void clickPayGrade() {
+        click(By.xpath(adminMenuPath));
+        click(By.xpath(jobTopMenuPath));
+        click(By.xpath(payGradeDropDownMenuPath));
+    }
+
+    public void clickPayGradeAddButton() {
+        click(By.xpath(addButtonPath));
+    }
+
+    public void setPayGrade(String payGradeName) {
+        setText(By.xpath(payGradeNameTextboxPath), payGradeName);
+    }
+
+    public Boolean savePayGrade() {
+        click(By.xpath(saveButtonPath));
+        return isElementVisible(By.xpath(successMessagePath));
+    }
+
+    public String checkPayGradeNameFieldIsMandatory() {
+        click(By.xpath(saveButtonPath));
+        isElementVisible(By.xpath(requiredFieldPath));
+        return getText(By.xpath(requiredFieldPath));
+    }
+
+    public void cancelButton() {
+        click(By.xpath(cancelButtonXpath));
+    }
+
+    public int totalNoOfPayGrades() {
+        Boolean isNoOFRecordsVisible = isElementVisible(By.xpath(noOfRecordsXpath));
+        if (isNoOFRecordsVisible == Boolean.TRUE) {
+            String noOfRecordsValue = getText(By.xpath(noOfRecordsXpath));
+            return noOfRecords(noOfRecordsValue);
+        }
+        return 0;
+    }
+
+    public boolean deleteButton(String name) {
+        isElementVisible(By.xpath("//i[@class=\"oxd-icon bi-pencil-fill\"]"));
```

Comment:
incorporate xpath as a variable in page class

Reply:
incorporated the xpath as a variable in page class.